core/prisma/migrations/20241205231134_backfill_default_forms/migration.sql

```diff
@@ -0,0 +1,85 @@
+WITH "components" AS (
+    SELECT
+        "c".*
+    FROM (
+        VALUES 
+        ('Boolean'::"CoreSchemaType", 'checkbox'::"InputComponent"),
+        ('String'::"CoreSchemaType", 'textInput'::"InputComponent"),
+        ('DateTime'::"CoreSchemaType", 'datePicker'::"InputComponent"),
+        ('Number'::"CoreSchemaType", 'textInput'::"InputComponent"),
+        ('NumericArray'::"CoreSchemaType", 'multivalueInput'::"InputComponent"),
+        ('StringArray'::"CoreSchemaType", 'multivalueInput'::"InputComponent"),
+        ('Email'::"CoreSchemaType", 'textInput'::"InputComponent"),
+        ('FileUpload'::"CoreSchemaType", 'fileUpload'::"InputComponent"),
+        ('URL'::"CoreSchemaType", 'textInput'::"InputComponent"),
+        ('MemberId'::"CoreSchemaType", 'memberSelect'::"InputComponent"),
+        ('Vector3'::"CoreSchemaType", 'confidenceInterval'::"InputComponent"),
+        ('Null'::"CoreSchemaType", NULL::"InputComponent"),
+        ('RichText'::"CoreSchemaType", 'richText'::"InputComponent")
+    ) AS c("schema", "component")
+),
+"form" AS (
+    INSERT INTO
+        "forms"(
+            "name",
+            "pubTypeId",
+            "slug",
+            "communityId",
+            "isDefault"
+        )
+    SELECT
+        "pub_types"."name" || ' Editor (Default)',
+        "pub_types"."id",
+        -- Rough equivalent to our slugify js function, adapted from https://gist.github.com/abn/779166b0c766ce67351c588489831852
+        trim(
+            BOTH '-'
+            FROM
+                regexp_replace(
+                    lower(trim("pub_types"."name")),
+                    '[^a-z0-9\\-_]+',
+                    '-',
+                    'gi'
+                )
+        ) || '-default-editor',
+        "pub_types"."communityId",
+        TRUE
+    FROM "pub_types"
+    ON CONFLICT DO NOTHING
```

Comment:
Unless one of us intentionally creates a form on prod to cause a conflict here, this shouldn't be necessary since I've gone with a kind of ugly slug for the default form: `<pub_type>-default-editor`. But it's handy for testing that this doesn't fail if there's already an existing default form!

Reply:
i think that's fine, we only need to run this backfill once and i don't foresee us already having existing `pub_type-default-editor` slugged forms!

Reply:
>But it's handy for testing that this doesn't fail if there's already an existing default form!

I phrased this poorly. I meant that this on conflict statement makes it easier to test the migration because you can safely rerun it even if there are already some default forms.

But yes, I agree! Shouldn't be risky!